### ISSUE TITLE
Attach Mapillary token to query string

### ISSUE THREAD

**Issue by cbeddow** [openstreetmap/iD#6399]
We are looking at letting users login with Mapillary to access organizations and data. We'd want to have user click `sign-in`, sign-in in mapillary in the same tab with oAuth, then redirect to `https://www.openstreetmap.org/edit?token=<token>#map=<zoom>/<lon>/<lat>`.

This will allow the token to be retained after login so users can filter imagery by organization and access the Mapillary data API

**Comment by kratico** [comment]
@cbeddow I would rephrase to

Enable authenticated requests to the Mapillary API so user can access their organizations public/private imagery.

An authenticated request requires an auth token that can be obtained with an OAuth flow to Mapillary.

The OAuth flow requires a redirect url to process the Mapillary auth token in OSM.

**Comment by cbeddow** [comment]
@bhousel do you have an idea of what we would need to implement inside iD to make this work?

**Comment by bhousel** [comment]
> @bhousel do you have an idea of what we would need to implement inside iD to make this work?

Not yet, sorry.. I expect for iD v3 all of Mapillary will move into a plugin, but we don't know what those will look like yet.

**Comment by cbeddow** [comment]
Our oAuth looks like this, for example with an ArcGIS app:

https://www.mapillary.com/connect?scope=private:read&client_id=XYZ123&redirect_uri=http://cbeddow:3344/webappbuilder/apps/44/widgets/Mapillary/oauth-callback.html&state=return&response_type=token

So once visiting that link and clicking authorize, it redirects to the redirect URI, with `token=11002233444` for example in the redirect query string.

So what we'd need is similar to the `photo_overlay=mapillary` query string feature that allows the Mapillary layer to be turned on if that's in the URL, but instead have a variable called `var mapillary_token;` in the OSM iD code which detects if a redirect comes through with the token in the string, so the token in the redirect enters the code. So it would be null my default it seems, which is fine, and then when populated with a token it then would trigger a few events that we can handle in our own development.

So overall the missing link would be how to get that token from the redirect URL I think. @kratico let me know if that sounds more clear?

@quincylvania does this sound like it would be similar implementation to the street level imagery layer in the query string?

**Comment by kratico** [comment]
@cbeddow it sounds clear

The problem is where to redirect and once redirected how to pass or store the token.

A few alternatives

- redirect to OSM iD, if `querystring.has('access_token')` then save token and reload the app (otherwise the token will remain visible in the URL)
- from OSM iD open OAuth login in a second tab (with `window.open`), redirect to OSM iD (or some other URI with the same domain as OSM iD), if `querystring.has('access_token')` do a `window.opener.postMessage()` to communicate the token to the tab the opened the OAuth login and close second tab

**Comment by cbeddow** [comment]
We are going to solve this in a different more simple way, closing
